### ISSUE TITLE
B009 safe fix discards evaluation of ** arguments

### ISSUE THREAD

**Issue by linhphamhoa39-ui** [astral-sh/ruff#28732]
### Summary

I'm not sure whether this is intended, but B009's safe fix removes the evaluation of a `**` argument expression in the example below, changing the behavior of a program that originally runs successfully.

### Reproduction

Tested with Ruff 0.16.8, CPython 3.12.3, on Linux x86_64.

Save the following as `side_effect.original.py`:

```python
events = []
getattr(1, "real", **(events.append("kw") or {}))
print(events)
```

Run the following commands in Bash. The diagnostic-only `ruff check` command is expected to exit with code 1 because it reports B009; continue with the remaining commands.

```bash
cp side_effect.original.py side_effect.fixed.py

python3 side_effect.original.py

ruff check --isolated --select B009 --no-cache --target-version py312 \
    --no-unsafe-fixes --no-preview --output-format json side_effect.original.py

ruff check --isolated --select B009 --no-cache --target-version py312 \
    --no-unsafe-fixes --no-preview --output-format json --fix side_effect.fixed.py

python3 side_effect.fixed.py
```

### Observed behavior

The original program prints:

```text
['kw']
```

The B009 diagnostic reports `"applicability": "safe"`. Applying the fix changes the file to:

```python
events = []
(1).real
print(events)
```

The modified program prints:

```text
[]
```

Both Python programs exit successfully with code 0. Unsafe fixes were explicitly disabled.

### Expected behavior

My understanding is that a safe fix should preserve the evaluation of the argument expression and its side effects.

Here, `events.append("kw") or {}` modifies the list before producing an empty dictionary. Although unpacking that dictionary passes no additional keyword arguments to `getattr`, evaluating the expression still has an observable effect.

The replacement removes that evaluation entirely. I think it would be more appropriate to leave the call unchanged, or avoid offering this replacement as a safe fix when the argument expression cannot be discarded without changing behavior.

What concerns me is that the modified program still runs successfully: there is no exception to indicate that an operation has disappeared. I do not currently have a production incident or an affected real-world project to report; this is a controlled reproduction.

### Additional example: exception behavior also changes

The same tested environment shows a related change for:

```python
print(getattr(1, "real", extra=0))
```

Originally, this raises:

```text
TypeError: getattr() takes no keyword arguments
```

Using the same Ruff options and applying the B009 fix changes it to:

```python
print((1).real)
```

The modified program prints `1` and exits successfully instead of raising `TypeError`.

The first example is the main concern, since it demonstrates changed behavior even when the original call succeeds.

### AI assistance

Codex executed and recorded these reproductions, including the diagnostics, the files actually rewritten by Ruff, and the before/after Python outputs. I have not independently rerun the commands manually. AI also assisted with preparing this description.

### Version

ruff 0.16.8

**Comment by ntBre** [comment]
Thanks for the report! This does seem like a bug. As Micha said on #28745, I think it makes sense not to offer a fix in such cases, beyond just marking it unsafe. We could even consider not emitting a diagnostic since the valid arguments for `getattr` are positional-only ([docs](https://docs.python.org/3/builtins/functions.html#getattr)). Any keyword arguments, with side effects or not, indicate an invalid call.

We should fix the bug regardless, but I'd be curious if you hit this in real code. The example here seems pretty unusual.

**Comment by linhphamhoa39-ui** [comment]
Thanks for confirming! I didn't encounter this in a real-world project. I've recently been experimenting with auditing development tools, and Codex found this issue during targeted testing. We also did a limited search of public source code, but haven't identified any affected real-world examples so far.

I agree that this is an unusual example. When it's unclear how to preserve the original behavior, I think it's better not to offer an automatic fix.
